docs/gherkin.md

```diff
@@ -0,0 +1,211 @@
+
+Read more
+
+ * [Feature Introduction](gherkin.md#feature) – general structure of a feature
+ * [Given/When/Then (Steps)](gherkin.md#steps)
+
+## Feature Introduction {#feature}
+
+Every `.feature` file conventionally consists of a single feature. A line starting with the keyword Feature followed by free indented text starts a feature. A feature usually contains a list of scenarios. You can write whatever you want up until the first scenario, which starts with the word Scenario (or localized equivalent; Gherkin is localized for [dozens of languages](https://github.com/cucumber/cucumber/wiki/Spoken-languages)) on a new line. You can use tagging to group features and scenarios together independent of your file and directory structure.
+
+Every scenario consists of a list of steps, which must start with one of the keywords **Given, When, Then, But** or **And**. Cucumber treats them all the same, but you shouldn’t. Here is an example:
+
+```
+Feature: Serve coffee
+  Coffee should not be served until paid for
+  Coffee should not be served until the button has been pressed
+  If there is no coffee left then money should be refunded
+
+  Scenario: Buy last coffee
+    Given there are 1 coffees left in the machine
+    And I have deposited 1$
+    When I press the coffee button
+    Then I should be served a coffee
+```
+
+In addition to a scenario, a feature may contain a background, scenario outline and examples. Respective keywords (in English) and places to read more about them are listed below. You can get a list of localized keywords with cucumber `--i18n [LANG]`.
+
+|     keyword     |     localized           | more info see     |
+|-----------------|-------------------------|-------------------|
+| name            | 'English'               |                   |
+| native          | 'English'               |                   |
+| encoding        | 'UTF-8'                 |                   |
+| feature         | 'Feature'               | [Feature](gherkin.md#feature)
+| background      | 'Backgroound'           | [Background](gherkin.md#background)
+| scenario        | 'Scenario'              | [Scenario](gherkin.md#feature)
+| scenario_outline| 'Scenario Outline'      | [Scenario Outline](gherkin.md#background)
+| examples        | 'Examples'/ 'Scenarios' | [Scenario Outline](docs/gherkin.md#background)
+| given           | 'Given'                 | [Given/When/Then (Steps)](docs/gherkin.md#steps)
+| when            | 'When'                  | [Given/When/Then (Steps)](docs/gherkin.md#steps)
+| then            | 'Then'                  | [Given/When/Then (Steps)](docs/gherkin.md#steps)
+| and             | 'And'                   | [Given/When/Then (Steps)](docs/gherkin.md#steps)
+| but             | 'But'                   | [Given/When/Then (Steps)](docs/gherkin.md#steps)
+
+## Step definitions {#steps}
```

Comment:
Step definitions don't belong in this document. Please move to a separate step-definitions.md doc, in a saparete PR.